pkg/build/generator/generator.go

```diff
@@ -353,6 +363,13 @@ func (g *BuildGenerator) Clone(ctx kapi.Context, request *buildapi.BuildRequest)
 	newBuild := generateBuildFromBuild(build, buildConfig)
 	glog.V(4).Infof("Build %s/%s has been generated from Build %s/%s", newBuild.Namespace, newBuild.ObjectMeta.Name, build.Namespace, build.ObjectMeta.Name)
 
+	// Add the original Build's OwnerReferences to the new Builds. This will
+	// make sure that the new Build is deleted when the original BuildConfig
+	// is deleted.
+	if len(build.OwnerReferences) > 0 {
```

Comment:
i'm not sure i'd classify it as intentional logic but yes that's the behavior and it's certainly worth noting.  I'm inclined to say it's acceptable behavior... us re-attaching the reference by looking up the associated buildconfig, if anything, seems to be contrary to what the user probably intended if they removed the ownerreference.